### ISSUE TITLE
Support for rebase merging

### ISSUE THREAD

**Issue by TimonVS** [toolmantim/release-drafter#106]
Currently rebase merging is not supported because Release Drafter checks for merge commits which don't exist when using rebase merging. At our company we're unable to use Release Drafter because we use rebase merging. Adding support for rebase merging requires a bit of a rewrite since it shouldn't rely on commits anymore, instead we can use the search endpoint (or search field in the GraphQL API) to fetch pull requests.

Here's an example query:

```gql
{
  search(type: ISSUE, query: "type:pr merged:>=2018-11-10 repo:toolmantim/release-drafter", first: 10) {
    nodes {
      ... on PullRequest {
        title
        number
        author {
          login
        }
        mergedAt
      }
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
}
```

Would you be interested in this @toolmantim? I could start working on it tomorrow and next week :)

**Comment by toolmantim** [comment]
Hi @TimonVS! We're actually hitting API limits with the current method (if there's lots of PRs to fetch), so switching to GraphQL would be better. Would be happy to switch to this.

If we could ensure that #98 is still possible though (adding directly-pushed-to-master commits), it'd be much appreciated.

**Comment by TimonVS** [comment]
It should drastically reduce the amounts of requests made. Currently I'm not sure if #98 will be possible with this approach, but I think it'd be a very useful feature to have. I've asked help for this in the Probot Slack :)

**Comment by toolmantim** [comment]
For #98, perhaps you might need to query both the commits and the PRs? Either separately, or in the same query?

**Comment by TimonVS** [comment]
That could work for merge commits and squash merging, but for rebase merging I will not be able to discern commits to master and commits that belong to a rebase merge. GitHub does however have a way to link the commits of a rebase merge to the PR as can be seen in the interface:
<img width="186" alt="pr-number" src="https://user-images.githubusercontent.com/876666/50375745-dda83380-0602-11e9-99b9-633fb2f17190.png">
But I'm unsure if they expose it in the API in some way (I couldn't find it). I haven't received a response on the Probot Slack yet, I will try to reach out to the API team directly.

**Comment by toolmantim** [comment]
Ah… I see! I've spotted `associatedPullRequests` on `Ref` in the GraphQL API before, but I'm guessing it doesn't do what we need.

**Comment by TimonVS** [comment]
Just wanted to let you know that I haven't forgotten about this. I've posted a question to the community forum here: https://github.community/t5/GitHub-API-Development-and/Get-pull-request-associated-with-a-commit/m-p/16764 but there's no solution yet.

**Comment by TimonVS** [comment]
It's been quite awhile, but it seems that GitHub has recently added a new field which makes it possible to retrieve the associated pull requests from a commit: https://developer.github.com/v4/changelog/2019-03-08-schema-changes/. I'll get on this tomorrow.

**Comment by toolmantim** [comment]
🎉

**Comment by TimonVS** [comment]
This should be solved now with #164 being merged :)
